sklearn/tests/test_props.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pytest
+
+from sklearn.feature_selection import SelectKBest
+from sklearn.pipeline import make_pipeline
+from sklearn.base import BaseEstimator, ClassifierMixin, TransformerMixin
+from sklearn.utils.validation import _validate_required_props
+from sklearn.datasets import make_classification
+from sklearn.metrics import make_scorer, balanced_accuracy_score
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+
+
+class MyEst(ClassifierMixin, BaseEstimator):
+    def __init__(self, C=1.0):
+        self._metadata_request = {'fit': ['sample_weight', 'brand']}
+        self.C = C
+
+    def fit(self, X, y, **fit_params):
+        _validate_required_props(self.get_metadata_request().fit, fit_params)
+        assert set(fit_params.keys()) <= \
+            set(self.get_metadata_request().fit.values())
+        self.svc_ = SVC(C=self.C).fit(X, y)
+        return self
+
+    def predict(self, X):
+        return self.svc_.predict(X)
+
+    def predict_proba(self, X):
+        return self.svc_predict_proba(X)
+
+
+class MyTrs(TransformerMixin, BaseEstimator):
+    def __init__(self):
+        self._metadata_request = {'fit': ['sample_weight']}
+
+    def fit(self, X, y=None, **fit_params):
+        req_props = self.get_metadata_request().fit
+        _validate_required_props(req_props, fit_params)
+        self._estimator = SelectKBest().fit(X, y)
+        assert set(fit_params.keys()) <= set(req_props.values())
+        return self
+
+    def transform(self, X, y=None):
+        return self._estimator.transform(X)
+
+
+def my_scorer(y, y_pred, **kwargs):
+    assert set(kwargs.keys()) == set(["new_param"])
```

Comment:
```suggestion
def my_scorer(y, y_pred, new_param):
```

Reply:
and maybe best to call this a "metric" not a "scorer" so as not to confuse APIs.